### PR TITLE
Fix incorrect exception message

### PR DESCRIPTION
**Pull request by wmtan**
There is an existing exception message designed for the case where the reuse of a process name results in a duplicate product of the same type with the same input tag.
Unfortunately, the same exception is thrown when the duplication is a result of the same product being registered more than once in the current process. In this second case, the exception message is not only misleading, but flat out wrong.
This PR distinguishes the two cases, and adds a new exception to be thrown in the second case.
This PR solves issue #13268

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: DataFormats/Provenance/src/ProductRegistry.cc [hunk 1]
```diff
@@ -81,9 +81,18 @@ namespace edm {
...
     if(!ret.second) {
-      throw Exception(errors::Configuration, "Duplicate Process")
-        << "The process name " << productDesc.processName() << " was previously used on these products.\n"
-        << "Please modify the configuration file to use a distinct process name.\n";
+      auto const& previous = *productList_.find(BranchKey(productDesc));
+      if(previous.second.produced()) {
+        // Duplicate registration in current process
+        throw Exception(errors::LogicError , "Duplicate Product")
+          << "The produced product " << previous.second << " is registered more than once.\n"
+          << "Please remove the redundant 'produces' call(s).\n";
```
Comment (Dr15Jones) [inline DataFormats/Provenance/src/ProductRegistry.cc]: Will the exception message contain the name of the module?

**Comment by Dr15Jones** [comment]
@wmtan ping

**Comment by wmtan** [comment]
@Dr15Jones Actually, no, but it prints out the module label and a lot of other info, using this existing function.

 void
  BranchDescription::write(std::ostream& os) const {
    os << "Branch Type = " << branchType() << std::endl;
    os << "Process Name = " << processName() << std::endl;
    os << "ModuleLabel = " << moduleLabel() << std::endl;
    os << "Branch ID = " << branchID() << '\n';
    os << "Class Name = " << fullClassName() << '\n';
    os << "Friendly Class Name = " << friendlyClassName() << '\n';
    os << "Product Instance Name = " << productInstanceName() << std::endl;
  }
I can add the module name either just to this exception, or more generally to the write() function.
Do you care which?

**Comment by wmtan** [comment]
@Dr15Jones ping Can you sign this?

**Comment by Dr15Jones** [comment]
+1

**Pull request merged**